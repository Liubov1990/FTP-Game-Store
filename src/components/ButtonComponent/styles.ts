export const buttonStyles = () => ({
  buttonRoot: {
    minWidth: "120px",
    height: "38px",
    fontFamily: "'Iceland', cursive",
    color: "#ffffff",
    backgroundColor: "#8b0000",
    textDecoration: "none",
    boxShadow: "none",
    "&:hover": {
      color: "#8b0000",
      textDecoration: "none"
    }
  },
  buttonPadded: {
    marginLeft: "30px"
  },

  "@media(max-width: 576px)": {
    buttonPadded: {
      marginLeft: 0,
      marginRight: "30px"
    }
  }
});
