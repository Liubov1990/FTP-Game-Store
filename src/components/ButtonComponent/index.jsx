import React from "react";
import { Button } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import cn from "classnames";
// styles
import { buttonStyles } from "./styles";

const useStyles = makeStyles(buttonStyles);

function ButtonComponent({ value, disabled, padded, ...rest }) {
  const classes = useStyles();

  return (
    <Button
      variant="contained"
      disabled={disabled}
      classes={{
        root: cn(classes.buttonRoot, {
          [classes.buttonPadded]: padded
        })
      }}
      {...rest}
    >
      {value}
    </Button>
  );
}
export default ButtonComponent;
