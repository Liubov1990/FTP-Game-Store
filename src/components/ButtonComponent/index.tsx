import React, { ReactElement } from "react";
import { Button, ButtonProps } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import cn from "classnames";
// styles
import { buttonStyles } from "./styles";

const useStyles = makeStyles(buttonStyles);

interface IButtonComponentProps extends ButtonProps {
  padded?: boolean,
}

function ButtonComponent({ value, disabled, padded, ...rest }: IButtonComponentProps): ReactElement {
  const classes = useStyles();

  return (
    <Button
      variant="contained"
      disabled={disabled}
      classes={{
        root: cn(classes.buttonRoot, {
          [classes.buttonPadded]: padded
        })
      }}
      {...rest}
    >
      {value}
    </Button>
  );
}
export default ButtonComponent;
