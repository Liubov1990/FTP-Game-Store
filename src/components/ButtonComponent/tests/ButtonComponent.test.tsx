import { render } from "@testing-library/react";
import ButtonComponent from "../index";

describe("ButtonComponent:", () => {
    test("renders the component", () => {
        const mockedProps = {value: "test button", disabled: false, padded: false};
        const { getByText, container } = render(<ButtonComponent {...mockedProps} />);
        expect(getByText(/test button/i)).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    })
})