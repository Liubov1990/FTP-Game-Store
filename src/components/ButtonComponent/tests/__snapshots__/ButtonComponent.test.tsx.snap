// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent: renders the component 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root makeStyles-buttonRoot-1 MuiButton-contained"
  tabindex="0"
  type="button"
>
  <span
    class="MuiButton-label"
  >
    test button
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;
